File: core/src/main/java/forpleuvoir/dhwu1a/core/event/request/BotInvitedJoinGroupRequestEvent.java
package forpleuvoir.dhwu1a.core.event.request;

import com.google.gson.annotations.SerializedName;

import static forpleuvoir.dhwu1a.core.common.ApiKey.GROUP_NAME;

/**
 * Bot被邀请入群申请
 *
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.event.request
 * <p>#class_name BotInvitedJoinGroupRequestEvent
 * <p>#create_time 2021/7/4 2:02
 */
public class BotInvitedJoinGroupRequestEvent extends RequestEvent {
    /**
     * 被邀请进入群的群名称
     */
    @SerializedName(GROUP_NAME)
    public final String groupName;

    private BotInvitedJoinGroupRequestEvent(Long eventId, Long fromId, String nick,
                                            String message,
                                            Long groupId,
                                            String groupName
    ) {
        super(RequestEventType.BotInvitedJoinGroupRequestEvent, eventId, fromId, groupId, nick, message);
        this.groupName = groupName;
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/event/request/BotInvitedJoinGroupRequestEvent.kt
package forpleuvoir.dhwu1a.core.event.request

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.GROUP_NAME

/**
 * Bot被邀请入群申请
 *
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.event.request
 *
 * #class_name BotInvitedJoinGroupRequestEvent
 *
 * #create_time 2021/7/4 2:02
 */
class BotInvitedJoinGroupRequestEvent private constructor(
    eventId: Long, fromId: Long, nick: String,
    message: String,
    groupId: Long,
    /**
     * 被邀请进入群的群名称
     */
    @field:SerializedName(GROUP_NAME) val groupName: String
) : RequestEvent(RequestEventType.BotInvitedJoinGroupRequestEvent, eventId, fromId, groupId, nick, message)